tasks/src/lib.rs

```diff
@@ -120,6 +267,34 @@ impl Scheduler {
         TaskHandle::new(handle)
     }
 
+    pub async fn exec_task<T>(&mut self, task: T) -> Result<T::Output, SchedulerError>
```

Comment:
Simple wrapper around `enqueue_task` that takes a task, enqueues it then awaits for its resolution.